### PR TITLE
Fixed flaky test

### PR DESCRIPTION
**Pull request by nathannaveen**
# Description of the PR

* Some parts of `Test_githubClient_GetWorkflow` were flaky, and some parts of the test would always fail.

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: internal/client/githubclient/githubclient_test.go [hunk 1]
```diff
@@ -437,11 +437,11 @@ func Test_githubClient_GetWorkflow(t *testing.T) {
 				return
 			}
 			if len(got) != len(test.want) {
-				t.Fatalf("GetWorkflow got %v, want %v", got, test.want)
+				t.Fatalf("GetWorkflow() got = %v, want %v", err, test.want)
```
Comment (pxp928) [inline internal/client/githubclient/githubclient_test.go:440]: why is this set to `err` should it not be `got`?

File: internal/client/githubclient/githubclient_test.go [hunk 2]
```diff
@@ -437,11 +437,11 @@ func Test_githubClient_GetWorkflow(t *testing.T) {
...
 			}
-			for i := 0; i < len(got); i++ {
+			for i := range got {
 				if got[i].Name != test.want[i].Name {
-					t.Fatalf("GetWorkflow got %v, want %v", got[i].Name, test.want[i].Name)
+					t.Errorf("GetWorkflow() got = %v, want %v", got, test.want)
```
Comment (pxp928) [inline internal/client/githubclient/githubclient_test.go:444]: why this change from `Fatalf` to `Errorf`?

File: internal/client/githubclient/githubclient_test.go (hunk 1 above)
Comment (pxp928) [reply internal/client/githubclient/githubclient_test.go:440]: should be length does not match error

Comment (nathannaveen) [reply internal/client/githubclient/githubclient_test.go]: Oh, I forgot to change this during rebase. Thanks!

File: internal/client/githubclient/githubclient_test.go (hunk 2 above)
Comment (nathannaveen) [reply internal/client/githubclient/githubclient_test.go]: Also forgot to change this during the rebase.

File: internal/client/githubclient/githubclient_test.go (hunk 1 above)
Comment (nathannaveen) [reply internal/client/githubclient/githubclient_test.go]: Updated to a length not matching error.

File: internal/client/githubclient/githubclient_test.go [hunk 3]
```diff
@@ -437,11 +437,11 @@ func Test_githubClient_GetWorkflow(t *testing.T) {
...
 			}
 			if len(got) != len(test.want) {
-				t.Fatalf("GetWorkflow got %v, want %v", got, test.want)
+				t.Fatalf("GetWorkflow() got length doesn't match want length = %v, want %v", len(got), len(test.want))
 			}
-			for i := 0; i < len(got); i++ {
+			for i := range got {
 				if got[i].Name != test.want[i].Name {
-					t.Fatalf("GetWorkflow got %v, want %v", got[i].Name, test.want[i].Name)
+					t.Fatalf("GetWorkflow() got = %v, want %v", got, test.want)
```
Comment (jeffmendoza) [inline internal/client/githubclient/githubclient_test.go:444]: The old line looks more correct here `got[i].Name, test.want[i].Name)`

Comment (jeffmendoza) [reply internal/client/githubclient/githubclient_test.go:444]: those are the two things being compared

Comment (nathannaveen) [reply internal/client/githubclient/githubclient_test.go]: Good idea, reverted back to the old error message.

**Review by pxp928 — APPROVED** [review]
